file-formats/uipg/type/zif_aff_uipg_v1.intf.abap

```diff
@@ -0,0 +1,201 @@
+INTERFACE zif_aff_uipg_v1
+  PUBLIC.
+
+  TYPES:
+    "! <p class="shorttext">General Information</p>
+    "! General information
+    BEGIN OF ty_general_information,
+
+      "! <p class="shorttext">Title</p>
+      "! Title visible to the enduser
+      "! $required
+      title       TYPE zif_aff_types_v1=>ty_description_100,
+
+      "! <p class="shorttext">Description</p>
+      "! Description for design-time tools
+      "! $required
+      description TYPE zif_aff_types_v1=>ty_description_100,
+
+      "! <p class="shorttext">Merge ID</p>
+      "! Merge ID for merging during run-time
+      merge_id    TYPE c LENGTH 35,
+
+    END OF ty_general_information.
+
+
+  "! <p class="shorttext">Catalog Type</p>
+  "! Catalog type
+  TYPES ty_catalog_type        TYPE c LENGTH 3.
+
+
+  "! <p class="shorttext">Tile Display Format</p>
+  "! Tile display format
+  "! $values {@link zif_aff_uipg_v1.data:co_tile_display_format}
+  "! $default {@link zif_aff_uipg_v1.data:co_tile_display_format.standard}
+  TYPES ty_tile_display_format TYPE c LENGTH 2.
+
+
+  CONSTANTS:
+    "! <p class="shorttext">Tile Display Format</p>
+    "! Tile display format
+    BEGIN OF co_tile_display_format,
+
+      "! <p class="shorttext">Standard Tile</p>
+      "! Standard tile
+      standard  TYPE ty_tile_display_format VALUE 'TI',
+
+      "! <p class="shorttext">Wide Tile</p>
+      "! Wide tile
+      wide      TYPE ty_tile_display_format VALUE 'TW',
+
+      "! <p class="shorttext">Flat Tile</p>
+      "! Flat tile
+      flat      TYPE ty_tile_display_format VALUE 'FT',
+
+      "! <p class="shorttext">Flat Wide Tile</p>
+      "! Flat wide tile
+      flat_wide TYPE ty_tile_display_format VALUE 'FW',
+
+      "! <p class="shorttext">Link</p>
+      "! Link
+      link      TYPE ty_tile_display_format VALUE 'LK',
+
+    END OF co_tile_display_format.
+
+
+  TYPES:
+    "! <p class="shorttext">Catalog Item Reference</p>
+    "! Catalog item reference
+    BEGIN OF ty_catalog_item_reference,
+
+      "! <p class="shorttext">Catalog ID</p>
+      "! Catalog ID
+      "! $required
+      catalog_id   TYPE c LENGTH 35,
+
+      "! <p class="shorttext">Catalog Type</p>
+      "! Catalog type
+      "! $default 'CAT'
+      catalog_type TYPE ty_catalog_type,
+
+      "! <p class="shorttext">Item ID</p>
+      "! Item ID
+      "! $required
+      id           TYPE c LENGTH 50,
+
+    END OF   ty_catalog_item_reference.
+
+
+  TYPES:
+    "! <p class="shorttext">Tile Assignment</p>
+    "! Tile assignment
+    BEGIN OF ty_tile_assignment,
+
+      "! <p class="shorttext">Tile Display Format</p>
+      "! Describes how the tile is rendered during run-time
+      tile_display_format TYPE ty_tile_display_format,
+
+      "! <p class="shorttext">Tile Reference</p>
+      "! Tile reference
+      "! $required
+      tile                TYPE ty_catalog_item_reference,
```

Comment:
Usually I would expect required fields at the beginn of the structure.

Minor: Mismatch between title and field name. There are some more places. But I understood the link between these. Maybe, this is fine. Double check with @wurzka

Reply:
Of course, it is better when the name and the title match. But I agree with @schneidermic0, in this case the link is clear. So, it is fine for me.

Reply:
I made a slight change here that I feel works a bit better. 

I now refer to tile and target mapping as tile key and target mapping key as this is essentially what these properties are. They are not fully specified tiles/target mapping, but only pointers to them. This is what I wanted to express with the description and I think the new property name does express the same. 

In general I tried to simplify all the names by removing unnecessary parts of them. e.g. I don't call it Page Section any more, because this are structure for pages so it is clear what is meant based on the context.